### PR TITLE
Fix dark mode text color for new follow button, interest chip 

### PR DESCRIPTION
**Pull request by ashiagr**
This PR updates `colorOnSecondary` [from black to white in the dark mode](https://github.com/wordpress-mobile/WordPress-Android/blob/974f5f219ddc2c0680f48ce89e03ea7c360b518a/WordPress/src/main/res/values-night/styles.xml#L13). `colorSecondary` is a pink shade in both light + dark mode and black `colorOnSecondary` didn't look good on it.

cc @mbshakti 

`colorOnSecondary`  is used in the new follow button ([Figma design link](https://www.figma.com/file/IDwOB0lUVRDCVAQ80YNNCz/%F0%9F%93%90-Topic-Detail-Page?node-id=194%3A468)) and interest chip text color selectors. It is not used anywhere else in the app.

To test:

Prerequisites - Reader Improvements feature flag is enabled 

- Launch the app and switch to dark mode
- Go to Reader tab
- Notice that interest picker is shown 
- Click an interest chip
- Notice that selected chip text color is white

-------

- Launch the app and switch to dark mode
- Go to Reader => Topic/ Tag details page
- Notice that Follow button text color is white

|Before|After| 
|---|---|
| food_before | food_new| 
| interests_before | inetersts|

**Merge instructions**

- Review this PR
- Make sure #12478 is merged
- Update target branch to develop
- Remove the "Not ready for merge" label
- Merge this PR

PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by mbshakti** [comment]
Yay for this!

File: WordPress/src/main/res/values/reader_styles.xml [hunk 1]
```diff
@@ -211,7 +211,7 @@
 
     <!-- chips -->
     <style name="ReaderInterestFilterChip" parent="@style/Widget.MaterialComponents.Chip.Filter">
-        <item name="android:textColor">@color/on_surface_surface_selector</item>
+        <item name="android:textColor">@color/on_surface_on_secondary_selector</item>
```
Comment (malinajirka) [inline WordPress/src/main/res/values/reader_styles.xml:214]: I'm wondering if it's ok to use "secondary_selector" for a button which uses "primaryColor". Wdyt?

Comment (ashiagr) [reply WordPress/src/main/res/values/reader_styles.xml]: You're right, it doesn't seem ok. I checked we've `colorOnPrimarySurface` in addition to `colorOnPrimary`

`colorOnPrimary` => light mode = white, dark mode = black
`colorOnPrimarySurface` => light mode = white, dark mode = white

`colorOnPrimarySurface` serves our purpose so switching to it in 6b0efa3

Comment (malinajirka) [reply WordPress/src/main/res/values/reader_styles.xml]: Awesome, thanks!

**Pull request merged**